### PR TITLE
JENKINS-22283 Show upstream jobs

### PR DESCRIPTION
**Pull request by tommysdk**
Rebased [PR 213](https://github.com/Diabol/delivery-pipeline-plugin/pull/213) with current master and made some refactoring (extract method/classes) as well as added some unit test assertions.

### CODE REVIEW

File: src/main/java/se/diabol/jenkins/pipeline/domain/DownstreamPipeline.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    @Override
+    public List<Pipeline> createPipelineLatest(int noOfPipelines,
+                                               ItemGroup context,
+                                               boolean pagingEnabled,
+                                               boolean showChanges,
+                                               Component component) throws PipelineException {
+        List<AbstractProject> firstProjects = ProjectUtil.getStartUpstreams(getFirstProject());
+        List<AbstractBuild> builds = resolveBuilds(firstProjects);
+
+        //TODO check if in queue
```
Comment (tommysdk) [inline src/main/java/se/diabol/jenkins/pipeline/domain/DownstreamPipeline.java]: @patbos what were your thoughts here?

**Comment by tommysdk** [comment]
Previous test coverage concerns have been addressed. I think this feature is quite ready to be merged. @patbos Could you please test and review?

**Comment by bsideup** [comment]
Hey @tommysdk @patbos,
Any progress here?

**Comment by tommysdk** [comment]
@bsideup I've been waiting for @patbos feedback on this one but I consider this good to go so if there are no objections I plan to merge this soon.

**Comment by tommysdk** [comment]
Good to merge.

**Comment by KostyaSha** [comment]
You have 2 1.0.0 tags in gh, what is the difference? 
PS don't forget to publish tag in gh ;)

**Comment by tommysdk** [comment]
@KostyaSha release-1.0.0 is a tag created to trigger the automatic release of a new version of the Delivery Pipeline plugin. The other tag corresponds to the actual release made by the automated process.

**Pull request merged**